### PR TITLE
Add tests to the ExecutionContext class

### LINKED ISSUE #17702: Add unit/component test to the `ExecutionContext` class
| Component |
| --- |
| content_manager |

## Description

Currently, the "ExecutionContext" class lacks unit and component tests, which limits our ability to ensure its proper functionality. This issue aims to add a robust set of automated tests that cover the key functionalities of the class.

## Expected

- Define and write unit tests for all the main functions and methods of the class. 
- Develop component tests (if applicable) for critical parts of the class.
- These tests should verify the expected behavior and catch any potential errors or failures.
- Identify possible improvements in the implementation.

## DoD

- Unit tests have been written for all key functions and methods of the "ExecutionContext" class.
- Component tests (if applicable) have been developed for the significant parts of the class.
- The tests run successfully without errors and detect potential issues.

### PR DESCRIPTION
**Pull request by kevin-mcm**
|Related issue|
|---|
| Closes #17702 |

## Description

This PR aims to add unit tests to the ExecutionContext class. Component tests are not added because this class does not have dependencies on other classes. Additionally, the [removeOutputFolderIfExists](https://github.com/wazuh/wazuh/blob/2649c354f9e5857066599c37cff567f6e50ed4b2/src/shared_modules/content_manager/tests/unit/executionContext_test.cpp#L24) function was added so the tests will not depend on previous executions since these do not remove the folders and create conflicts with other tests.

### CODE REVIEW

**Comment by kevin-mcm** [comment]
# Tests - Execution Context
```Bash
[----------] 5 tests from ExecutionContextTest
[ RUN      ] ExecutionContextTest.instantiation
[       OK ] ExecutionContextTest.instantiation (0 ms)
[ RUN      ] ExecutionContextTest.TestDefaultFolderWhenThereIsNoConfigurationForTheOutputFolder
[       OK ] ExecutionContextTest.TestDefaultFolderWhenThereIsNoConfigurationForTheOutputFolder (0 ms)
[ RUN      ] ExecutionContextTest.TestDefaultFolderWhenTheOutputFolderPathIsEmpty
[       OK ] ExecutionContextTest.TestDefaultFolderWhenTheOutputFolderPathIsEmpty (0 ms)
[ RUN      ] ExecutionContextTest.TestValidCaseWhenTheOutputFolderPathIsNotEmpty
[       OK ] ExecutionContextTest.TestValidCaseWhenTheOutputFolderPathIsNotEmpty (0 ms)
[ RUN      ] ExecutionContextTest.TestValidCaseWhenTheOutputFolderPathIsNotEmptyAndExists
[       OK ] ExecutionContextTest.TestValidCaseWhenTheOutputFolderPathIsNotEmptyAndExists (0 ms)
[----------] 5 tests from ExecutionContextTest (1 ms total)
```

**Review by Damian-Mangold — COMMENTED** [review]
The use of `std::cout` is temporary and should be replaced by a log message, so I think using this message to verify the correct behavior is not the best because when we replace the use of `std::cout`, we will need to update the test as well.
I think a better approach is to check if the folder exists, or add some files in the folder and check if it has been deleted..... using a more generic strategy.

File: src/shared_modules/content_manager/tests/unit/executionContext_test.cpp [hunk 1]
```diff
@@ -47,17 +47,14 @@ TEST_F(ExecutionContextTest, TestDefaultFolderWhenThereIsNoConfigurationForTheOu
     // Remove the output folder if exists
     removeOutputFolderIfExists(GENERIC_OUTPUT_FOLDER_PATH);
 
-    const auto expectedCapturedOutput = "Created output folder: \"" + GENERIC_OUTPUT_FOLDER_PATH + "\"\n";
-
     m_spUpdaterBaseContext->configData.erase("outputFolder");
 
+    // Start silence stdout
     testing::internal::CaptureStdout();
```
Comment (Damian-Mangold) [inline src/shared_modules/content_manager/tests/unit/executionContext_test.cpp:53]: I understand that the goal of this is to have a cleaner test execution report. But, when we replace the std::cout with the logging mechanism we have to update the test by removing these lines. With this in mind, I would rather have, temporarily a report with some extra printing, than modify the tests in the future.

Comment (kevin-mcm) [reply src/shared_modules/content_manager/tests/unit/executionContext_test.cpp:53]: I will remove the silence from `std::cout`

Comment (kevin-mcm) [reply src/shared_modules/content_manager/tests/unit/executionContext_test.cpp]: done

**Pull request merged**